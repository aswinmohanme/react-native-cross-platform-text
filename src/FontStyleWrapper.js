import { StyleSheet } from 'react-native';
import getFontStyleForWeight from './getFontStyleForWeight';

const FontStyleWrapper = ({ style, children }) => {
  const flatStyle = StyleSheet.flatten(style);
  const fontStyleForWeight = getFontStyleForWeight(flatStyle.fontWeight, flatStyle.fontFamily);
  return children([flatStyle, fontStyleForWeight]);
};

export default FontStyleWrapper;
