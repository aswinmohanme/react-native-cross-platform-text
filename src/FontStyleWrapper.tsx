import * as React from "react";
import { StyleSheet, StyleProp, TextStyle } from "react-native";
import getFontStyleForWeight from "./getFontStyleForWeight";

interface FontStyleWrapperProps {
  children: (styles: TextStyle) => React.ReactElement<{}>;
  style: StyleProp<TextStyle>;
}

const FontStyleWrapper = ({ style, children }: FontStyleWrapperProps) => {
  const flatStyle = StyleSheet.flatten(style);
  const fontStyleForWeight = getFontStyleForWeight(flatStyle.fontFamily, flatStyle.fontWeight);
  const styles = Object.assign({}, flatStyle, fontStyleForWeight);
  return children(styles);
};

export default FontStyleWrapper;
