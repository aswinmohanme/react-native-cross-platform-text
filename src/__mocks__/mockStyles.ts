import { StyleSheet } from 'react-native';

const doMockStyles = StyleSheet.create({
  fontStyle: {
    fontWeight: 'normal',
    fontFamily: 'Arial',
  },
  section: {
    paddingHorizontal: 10,
  },
  container: {
    margin: 20,
  },
});

export default doMockStyles;
