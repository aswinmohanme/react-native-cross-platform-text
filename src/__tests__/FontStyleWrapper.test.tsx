import * as React from "react";
import { Platform, Text } from "react-native";
import * as ShallowRenderer from "react-test-renderer/shallow";
import FontStyleWrapper from "../FontStyleWrapper";
import mockStyles from '../__mocks__/mockStyles';

jest.mock("Platform");

it("iOS returns fontFamily and weight for normal", () => {
  Platform.OS = "ios";
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(
    <FontStyleWrapper style={mockStyles.fontStyle}>
      {(fontStyle) => <Text style={fontStyle} />}
    </FontStyleWrapper>,
  );
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it("Android returns fontFamily-Regular and weight = undefined for normal", () => {
  Platform.OS = "android";
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(
    <FontStyleWrapper style={mockStyles.fontStyle}>
      {(fontStyle) => <Text style={fontStyle} />}
    </FontStyleWrapper>,
  );
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it("flattens styles", () => {
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(
    <FontStyleWrapper
      style={[
        mockStyles.section,
        [mockStyles.container, mockStyles.fontStyle],
      ]}
    >
      {(fontStyle) => <Text style={fontStyle} />}
    </FontStyleWrapper>,
  );
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
