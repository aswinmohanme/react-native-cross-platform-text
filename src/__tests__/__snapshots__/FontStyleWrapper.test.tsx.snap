// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Android returns fontFamily-Regular and weight = undefined for normal 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "fontFamily": "Arial-Regular",
      "fontWeight": undefined,
    }
  }
/>
`;

exports[`flattens styles 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "fontFamily": "Arial-Regular",
      "fontWeight": undefined,
      "margin": 20,
      "paddingHorizontal": 10,
    }
  }
/>
`;

exports[`iOS returns fontFamily and weight for normal 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "fontFamily": "Arial",
      "fontWeight": "normal",
    }
  }
/>
`;
